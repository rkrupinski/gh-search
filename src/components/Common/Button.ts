import styled from "styled-components";
import { darken } from "polished";

export const Button = styled.button`
  box-sizing: border-box;
  height: 2rem;
  padding: 0 0.75rem;
  border: 0;
  border-radius: ${({ theme }) => theme.layout.borderRadius};
  font-size: 0.875rem;
  line-height: 2;
  color: ${({ theme }) => theme.colors.textAlt};
  background: ${({ theme }) => theme.colors.primary};
  appearance: none;
  transition: opacity ${({ theme }) => theme.animation.duration};

  &:disabled {
    opacity: 0.7;
  }

  &:not(:disabled) {
    cursor: pointer;

    :hover {
      background: ${({ theme }) => darken(0.1, theme.colors.primary)};
    }
  }
`;
