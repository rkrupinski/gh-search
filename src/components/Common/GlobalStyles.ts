import { createGlobalStyle } from "styled-components";
import { normalize } from "polished";

export const GlobalStyles = createGlobalStyle`
  ${normalize()}

  html {
    min-height: 100%;
    font: 100%/1.5 Roboto, sans-serif;
    background: ${({ theme }) => theme.colors.bgSecondary};
  }

  body {
    min-height: 100%;
  }

  h1,
  h2,
  h3 {
    line-height: 1.375;
    font-weight: 500;
    color: ${({ theme }) => theme.colors.textPrimary};
  }

  a {
    text-decoration: none;
    color: ${({ theme }) => theme.colors.link};

    &:hover {
      text-decoration: underline;
    }
  }
`;
