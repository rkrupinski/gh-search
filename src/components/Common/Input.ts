import styled from "styled-components";

export const Input = styled.input`
  box-sizing: border-box;
  height: 2rem;
  padding: 0 0.75rem;
  font-size: 0.875rem;
  line-height: 2;
  border: 0;
  border-radius: ${({ theme }) => theme.layout.borderRadius};
  color: ${({ theme }) => theme.colors.textPrimary};
  background: ${({ theme }) => theme.colors.bgSecondary};
  appearance: none;

  ::placeholder {
    color: ${({ theme }) => theme.colors.textSecondary};
  }
`;
