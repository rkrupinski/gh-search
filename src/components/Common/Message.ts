import styled, { keyframes } from "styled-components";

const appear = keyframes`
  to { opacity: 1; }
`;

const icons: Record<MessageType, string> = {
  loading: "‚è≥",
  error: "üí©",
};

export type MessageType = "loading" | "error";

export type MessageProps = {
  type: MessageType;
  waitMs?: number;
};

export const Message = styled.p<MessageProps>`
  margin: calc(${({ theme }) => theme.layout.gutter} * 3) 0;
  text-align: center;
  word-wrap: break-word;
  color: ${({ theme }) => theme.colors.textSecondary};
  opacity: 0;
  animation-name: ${appear};
  animation-fill-mode: forwards;
  animation-duration: ${({ theme }) => theme.animation.duration};
  animation-delay: ${({ waitMs }) => `${waitMs ?? 0}ms`};

  &:before {
    content: ${({ type }) => `"${icons[type]}"`};
    margin-right: 0.25rem;
  }
`;
