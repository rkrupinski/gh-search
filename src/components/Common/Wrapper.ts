import styled from "styled-components";

export const Wrapper = styled.div`
  box-sizing: border-box;
  max-width: ${({ theme }) => theme.layout.maxWidth};
  padding: 0 ${({ theme }) => theme.layout.gutter};
  margin: 0 auto;
`;
