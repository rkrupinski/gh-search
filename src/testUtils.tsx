import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { StoreProvider } from "store";
import { theme } from "theme";

const Providers: React.FC = ({ children }) => (
  <StoreProvider>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </StoreProvider>
);

export const renderWithProviders = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries" | "wrapper">,
) => render(ui, { wrapper: Providers, ...options });

export const noop = () => {
  /**/
};
