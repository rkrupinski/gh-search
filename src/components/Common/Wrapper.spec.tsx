import React from "react";
import { Wrapper } from "components/Common/Wrapper";
import { renderWithProviders } from "testUtils";

describe("Wrapper", () => {
  it("should render correctly", () => {
    const { container } = renderWithProviders(<Wrapper>Test</Wrapper>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
