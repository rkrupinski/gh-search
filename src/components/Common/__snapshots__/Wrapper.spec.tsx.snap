// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper should render correctly 1`] = `
.c0 {
  box-sizing: border-box;
  max-width: 42rem;
  padding: 0 1rem;
  margin: 0 auto;
}

<div
  class="c0"
>
  Test
</div>
`;
